traitsui/qt4/file_editor.py

```diff
@@ -48,16 +54,36 @@ def init(self, parent):
 
         if self.factory.auto_set:
             control.textEdited.connect(self.update_object)
+            self._connections_to_remove.append(
+                (control.textEdited, self.update_object)
+            )
         else:
             # Assume enter_set is set, or else the value will never get
             # updated.
             control.editingFinished.connect(self.update_object)
+            self._connections_to_remove.append(
+                (control.editingFinished, self.update_object)
+            )
 
         button = IconButton(QtGui.QStyle.SP_DirIcon, self.show_file_dialog)
+        # Call to connect should be moved out of IconButton
+        # (enthought/traitsui#888)
+        self._connections_to_remove.append(
+            (button.clicked, self.show_file_dialog)
+        )
```

Comment:
I think it might be better to move this inside `IconButton` and expose it as a separate method, because then the signal `clicked` is mentioned in the same place. I think the `IconButton` should behave like `QPushButton` and not try to connect signals at instantiation.
Moving the logic into `IconButton`...

Reply:
Now I remember why I didn't do it: It was because that requires `IconButton` to hold a reference to the editor's method, creating a reference cycle. 
On balance I am going to skip this disconnection because (1) `connect` and `disconnect` should be co-located, having them separate risk erroneous inconsistency in the future - attempt to disconnect an unconnected signal will cause Qt to raise an error and (2) having them co-located without modifying the public interface of `IconButton` will create reference cycle, and is not a very good pattern anyway.

This should be dealt within #888

Reply:
Should this line be removed then?

Reply:
Yes. Somehow I only managed to remove the comment 🤦 . Thank you!